sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -868,6 +868,8 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product with Tre
     case null => Nil
     case None => Nil
     case Some(null) => Nil
+    case Some(table: CatalogTable) =>
```

Comment:
This triggers test failures on SQLQueryTestSuite / ThriftServerQueryTestSuite due to the change of representation of CatalogTable wrapped with Some. While I think it "was" a bug or at least not intended (because we care for CatalogTable not wrapped with Some), please let me know if we want to retain the current.